vllm/v1/worker/gpu_input_batch.py

```diff
@@ -285,6 +288,28 @@ def add_request(
         if sampling_params.logit_bias is not None:
             self.logit_bias[req_index] = sampling_params.logit_bias
 
+        if sampling_params.allowed_token_ids:
+            # NOTE(houseroad): put the check here since no vocab_size info
+            # available in vllm/sampling_params.py
+            if not all(0 <= tid < self.vocab_size
+                       for tid in sampling_params.allowed_token_ids):
+                raise ValueError(
+                    "allowed_token_ids contains out-of-vocab token id")
```

Comment:
I think this will be a problem because it will cause the entire batch to fail.

Could you move the validation to [here](https://github.com/vllm-project/vllm/blob/main/vllm/v1/engine/processor.py#L103)?

Reply:
Yeah, good point. Moved.